products/platform/documentation-site/research/search-design-doc.md

```diff
@@ -1,15 +1,207 @@
+
+In general, this kind of system requires the following components:
+
+- Page with Input Textbox - We need a public page with a search input where users can input their search term(s).
+- Index/Database - We need a place to store the records that will be returned to the user.
+- Search API - We need a way to get data out of the index.
+- Crawler/Scraper - We need a crawler to scan our documentation sources and push data to our index.
+- Pipeline - We need a way to routinely run the crawler
+
+By using Algolia's Search API product with their open source Dropdown Search-UI and open source scraper, we can avoid building/maintaining the database, API, UI, and crawler components.
+
+That means we'd be responsible for:
+
+- Converting `va.gov-team`'s `docs/index.md` from markdown to HTML
+- Adding a search input to `va.gov-team`'s `docs/index.html`
+- Adding the DocSearch `<style>` and `<script>` tags to `va.gov-team`'s `docs/index.html`
+- Writing the config file for the scraper
+- Adding CircleCI to `va.gov-team`
+- Adding a scheduled job to `va.gov-team`'s CircleCI pipeline to run the scraper once every 24 hours
+- Manually triggering the initial scraper run to populate our Algolia index
+
+#### Usage limits
+
+[Algolia's pricing page](https://www.algolia.com/pricing/) shows three different tiers, in addition to an enterprise plan. The primary differentiator between the tiers is the number of records and the number of operations (read/write/edit/delete) included with each plan.
+
+##### The Community Plan
+
+The free tier, the Community plan, includes up to 10K records and 50K operations (read/write/edit/delete). If we exceed the limits of the free tier, we'd have to upgrade to a paid tier.
+
+##### The Starter Plan
+
+The middle tier, the Starter plan, includes up to 50K records and 250k operations. If we exceed the limits of the Starter plan, it would cost an extra $10/month for every additional 100K operations, and an extra $10/month for every additional 20K records.
+
+##### The Pro Plan
+
+The upper tier, the Pro plan, includes up to 1M records and 5M operations. If we exceed the limits of the Pro plan, it would cost an extra $5/month foe every 100K additional operations, and an extra $5/month for every additional 20K records.
+
+---
+
+Essentially, every line of text (`<p>`, `<li>`, `<h1-6>`, etc) in a markdown file ends up as a separate record in the Algolia index. We have approximately 3K `.md` files in the `va.gov-team` repo. We ran a test crawler on the `va.gov-team` repo to estimate how many elements would match our selectors, and it found almost 150K matching elements, which equates to almost 150K records.
+
+#### Where/when to run the scraper
+
+Subject to review, we will run the crawler in a scheduled job once every 24 hours in a Circle CI pipeline for the `va.gov-team` repo.
+
+It's possible that running the scraper once per day would consume a significant number of operations.
+
+#### Private repos
```

Comment:
I feel like this is the real sticking point for me on any search product. If we index a private repo with a public search function, that information becomes public. Any ideas on how this could work?

Reply:
I think the answer here is "we're not going to index the private repos", but that should be explicitly documented in this design

Reply:
Explicitly documented in c57dc1119981f03d57062e6c027f9a7abece2872